onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2701,10 +2701,10 @@ def aten_index_put(
 
     index = op.SequenceAt(indices, 0)  # assume indices only have 1 element
     # change array([1,3]) to array([[1,1,1,1,1],[3,3,3,3,3]])
-    self_rank_1 = op.Gather(op.Shape(self), 1)
-    index_rank_0 = op.Gather(op.Shape(index), 0)
+    self_dim_1 = op.Gather(op.Shape(self), 1)
```

Comment:
Just out of curiosity: would Slice work too, or do we have to use Gather here?

Reply:
Slice need more steps than Gather. And if the starts = -1, you don't know how to set the ends value.

Reply:
I see. thanks!